src/canister_tests/src/tests.rs

```diff
@@ -761,6 +765,121 @@ mod delegation_tests {
         );
         Ok(())
     }
+
+    /// Verifies that get_principal and prepare_delegation return the same principal.
+    #[test]
+    fn get_principal_should_match_prepare_delegation() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+        let user_number = flows::register_anchor(&env, canister_id);
+        let frontend_hostname = "https://some-dapp.com";
+        let pub_session_key = ByteBuf::from("session public key");
+
+        let (canister_sig_key, _) = api::prepare_delegation(
+            &env,
+            canister_id,
+            principal_1(),
+            user_number,
+            frontend_hostname.to_string(),
+            pub_session_key.clone(),
+            None,
+        )?;
+
+        let principal = api::get_principal(
+            &env,
+            canister_id,
+            principal_1(),
+            user_number,
+            frontend_hostname.to_string(),
+        )?;
+        assert_eq!(Principal::self_authenticating(canister_sig_key), principal);
+        Ok(())
+    }
+
+    /// Verifies that get_principal returns different principals for different front end host names.
+    #[test]
+    fn should_return_different_principals_for_different_frontends() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+        let user_number = flows::register_anchor(&env, canister_id);
+        let frontend_hostname_1 = "https://dapp-1.com";
+        let frontend_hostname_2 = "https://dapp-2.com";
+
+        let dapp_principal_1 = api::get_principal(
+            &env,
+            canister_id,
+            principal_1(),
+            user_number,
+            frontend_hostname_1.to_string(),
+        )?;
+
+        let dapp_principal_2 = api::get_principal(
+            &env,
+            canister_id,
+            principal_1(),
+            user_number,
+            frontend_hostname_2.to_string(),
+        )?;
+
+        assert_ne!(dapp_principal_1, dapp_principal_2);
```

Comment:
I feel like I saw this exact check somewhere else.. but I think I'm dreaming

Reply:
There is also `should_issue_different_principals` for the `prepare_delegation` call. But I think both have their place because the implementation could be broken independently.